docs/source/poet/sgx_primer.rst

```diff
@@ -0,0 +1,205 @@
+attestation is used by PoET and Intel® provides an implementation of this service
+through the Intel® Attestation Service (IAS). It is also known as “remote
+attestation”.
+
+A special “Quoting Enclave” is provided with SGX and is used in Inter-process
+Attestation. The quoting enclave processes reports using the following
+procedure:
+
+* The Quoting Enclave verifies reports submitted to it from other enclaves on
+  the same platform using the Intra-process Attestation described above.
+* It then replaces the MAC with a signature created with a device specific
+  private key stored in the device hardware.
+* This new data structure is called a “quote”.
+
+The Quoting Enclave uses the Intel® Enhanced Privacy ID (EPID) signature scheme
+to overcome the privacy concerns associated with using a small number of
+asymmetric keys for signing over the life of a platform. “EPID is a group
+signature scheme that allows a platform to sign objects without uniquely
+identifying the platform or linking different signatures.” [2]_ In “anonymous”
+mode, a signature verifier cannot associate a given signature with a given
+identity. In “pseudonymous” mode, a signature verifier can determine whether it
+has verified a platform previously.
+
+The remote attestation procedure is roughly the following:
+
+1. A local application enclave wants to interact with a remote enclave so it
+   connects to the remote service through the local application.
+#. The remote service issues a challenge to the local application.
+#. The local application retrieves the local quoting enclave’s identity and
+   passes it into the the local application enclave along with the challenge.
+#. The local application enclave generates a “manifest” that includes:
+
+   a. A response to the challenge
+   #. An ephemeral public key
+
+#. The local application enclave generates a hash digest of the manifest. An
+   enclave report is generated which includes the hash digest and has the
+   platform’s (local) Quoting Enclave as its target enclave.
+#. The enclave report and manifest are sent to the local application which
+   forwards the report to the Quoting Enclave.
+#. The Quoting Enclave verifies the report, generates an enclave quote using its
+   EPID key, and forwards the quote to the local application.
+#. The local application forwards the enclave quote and manifest to the remote
+   service.
+#. The remote service verifies the quote and manifest
+
+   a. An EPID public key certificate along with revocation information can be
+      used to verify both locally
+   #. Alternatively, an attestation verification service can be used (see IAS
+      below).
+
+Intel® Attestation Service
+--------------------------
+
+As an alternative to keeping track of a set of verified and revoked enclave
+identities, Intel® provides an attestation service (IAS) which can be used to
+verify enclave quotes and challenge manifests. Roughly speaking, IAS provides:
```

Comment:
I deleted "and challenge manifest" because I might have gotten the wording wrong. My understanding and what I tried to convey in the remote attestation procedure is that when you request attestation from IAS, IAS sends you a challenge (some data) which needs to be signed by the enclave you are requesting attestation for.